File: android/app/src/main/java/tuerantuer/app/integreat/IntegreatPackage.java
package tuerantuer.app.integreat;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;
import tuerantuer.app.integreat.constants.NativeConstantsModule;
import tuerantuer.app.integreat.fetcher.FetcherModule;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class IntegreatPackage implements ReactPackage {

    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
        return Collections.emptyList();
    }

    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
        List<NativeModule> modules = new ArrayList<>();
        modules.add(new FetcherModule(reactContext));
        modules.add(new NativeConstantsModule(reactContext));
        return modules;
    }
}

File: native/android/app/src/main/java/tuerantuer/app/integreat/IntegreatPackage.kt
package tuerantuer.app.integreat

import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager
import tuerantuer.app.integreat.constants.NativeConstantsModule
import tuerantuer.app.integreat.fetcher.FetcherModule

class IntegreatPackage : ReactPackage {
    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
        return emptyList()
    }

    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
        val modules: MutableList<NativeModule> = ArrayList()
        modules.add(FetcherModule(reactContext))
        modules.add(NativeConstantsModule(reactContext))
        return modules
    }
}
